Simplenote/Classes/SPNoteListViewController+Extensions.swift

```diff
@@ -834,6 +834,30 @@ extension SPNoteListViewController {
     }
 }
 
+// MARK: - Excerpt
+//
+private extension SPNoteListViewController {
+
+    func bodyExcerpt(from note: Note, withKeywords keywords: [String]?) -> String? {
+        guard let keywords = keywords, !keywords.isEmpty, let content = note.content else {
+            return note.bodyPreview
+        }
+
+        guard let bodyRange = NoteContentHelper.structure(of: content).body else {
+            return note.bodyPreview
+        }
+        let excerpt = content.contentSlice(matching: keywords,
+                                           in: bodyRange,
+                                           leadingLimit: Constants.excerptLeadingLimit,
+                                           trailingLimit: Constants.excerptTrailingLimit)
+
+        let shouldAddEllipsis = excerpt.range.lowerBound > bodyRange.lowerBound
+        let excerptString = (shouldAddEllipsis ? "…" : "") + excerpt.normalized.content
+
+        return excerptString.replacingNewlinesWithSpaces()
+    }
```

Comment:
Question: (Thinking aloud!) do you think it'd make sense to move this API into a Note Extension?

```
extension Note {
    func bodyExcerpt(keywords: [String]?) -> String? { }
}
```

Reply:
+ Would it be possible to unit test this one?

Reply:
> do you think it'd make sense to move this API into a Note Extension?

Looks better, thank you!